### PR TITLE
LPD 18170 | Liferay form breaks if site URL has the word "shared"

### PR DESCRIPTION
**Pull request by Marina-Gouveia**
Related Issue: [LPD-18170](https://liferay.atlassian.net/browse/LPD-18170)
Related LPP: [LPP-52801](https://liferay.atlassian.net/browse/LPP-52801)

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
Hey @carolmariaabb 
Could you review this PR, please? :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java [hunk 1]
```diff
@@ -777,10 +777,20 @@ public Boolean isRequireAuthentication() throws PortalException {
...
+				themeDisplay.getURLCurrent(), "shared", StringPool.BLANK) &&
+			StringUtil.contains(
+				themeDisplay.getURLCurrent(),
+				String.valueOf(ddmFormInstance.getFormInstanceId()),
+				StringPool.BLANK)) {
+
+			return true;
+		}
+
+		return false;
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java:793]: @Marina-Gouveia please check if this change will impact the existing [test](https://github.com/liferay/liferay-portal/blob/16e66444b76c8bb1a39aba9a65327dba06746edd/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java#L668-L687)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java [hunk 2]
```diff
@@ -777,10 +777,20 @@ public Boolean isRequireAuthentication() throws PortalException {
...
-			themeDisplay.getURLCurrent(), "shared", StringPool.BLANK);
+		if (StringUtil.contains(
+				themeDisplay.getURLCurrent(), "shared", StringPool.BLANK) &&
+			StringUtil.contains(
+				themeDisplay.getURLCurrent(),
+				String.valueOf(ddmFormInstance.getFormInstanceId()),
+				StringPool.BLANK)) {
+
+			return true;
+		}
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java:791]: @Marina-Gouveia please change to:

```
ThemeDisplay themeDisplay = getThemeDisplay();

DDMFormInstance ddmFormInstance = getFormInstance();

if (StringUtil.contains(
		themeDisplay.getURLCurrent(), "shared", StringPool.BLANK) &&
	StringUtil.contains(
		themeDisplay.getURLCurrent(),
		String.valueOf(ddmFormInstance.getFormInstanceId()),
		StringPool.BLANK)) {

	return true;
}
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java [hunk 3]
```diff
@@ -362,7 +362,7 @@ public void testGetDDMFormContext() throws Exception {
 
 		ddmForm.addDDMFormField(ddmFormField);
 
-		Mockito.doReturn(
+		DDMForm ddmForm1 = Mockito.doReturn(
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java:365]: @Marina-Gouveia why do we need to assign to this variable? It is never used

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java [hunk 4]
```diff
@@ -671,18 +671,32 @@ public void testIsSharedURL() throws Exception {
...
 			ddmFormDisplayContext
 		).getThemeDisplay();
 
+		Mockito.doReturn(
+			ddmFormInstance
+		).when(
+			ddmFormDisplayContext
+		).getFormInstance();
+
 		Assert.assertTrue(ddmFormDisplayContext.isSharedURL());
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java:700]: @Marina-Gouveia please change to:

```
DDMFormInstance ddmFormInstance = Mockito.mock(DDMFormInstance.class);

Mockito.when(
	ddmFormInstance.getFormInstanceId()
).thenReturn(
	123L
);

Mockito.doReturn(
	ddmFormInstance
).when(
	ddmFormDisplayContext
).getFormInstance();

ThemeDisplay themeDisplay = Mockito.mock(ThemeDisplay.class);

Mockito.when(
	themeDisplay.getURLCurrent()
).thenReturn(
	"http://localhost:8080/web/forms/shared?form=123"
);

Mockito.doReturn(
	themeDisplay
).when(
	ddmFormDisplayContext
).getThemeDisplay();

Assert.assertTrue(ddmFormDisplayContext.isSharedURL());
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java (hunk 3 above)
Comment (Marina-Gouveia) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java]: Hi @carolmariaabb, I made the necessary adjustment, but it ended up going through inattention.

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:forms

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
@Marina-Gouveia the [LocalFile.Forms#GuestUserCanNotSeeFormAfterPermissionRemoved](https://testray.liferay.com/home/-/testray/case_results/2541364884/history?tab=result&hideFilterPin=true) is only failing in your PR. Could you please take a look?

**Comment by carolmariaabb** [comment]
@Marina-Gouveia please see https://github.com/liferay-forms/liferay-portal/pull/2478

**Pull request closed without merging**